tests/Unit/FunctionsTest.php

```diff
@@ -44,6 +47,29 @@ public function faker(): void
         $this->assertIsString(faker()->name());
     }
 
+    /**
+     * @test
+     */
+    public function lazy(): void
+    {
+        $callback = new LazyValue(fn () => new \stdClass());
+        $value = lazy($callback);
+
+        $this->assertInstanceOf(LazyValue::class, $value);
+        $this->assertNotSame($value(), $value());
+    }
+    /**
+     * @test
+     */
+    public function memoize(): void
+    {
+        $callback = new LazyValue(fn () => new \stdClass());
+        $value = memoize($callback);
+
+        $this->assertInstanceOf(LazyValue::class, $value);
+        $this->assertSame($value(), $value());
+    }
```

Comment:
These duplicate the LazyValueTest. But I feel like they're necessary to ensure the functions are wired in correctly.

Reply:
that's fine IMO